### PR TITLE
Add documentation for SpooledProducer interface

### PR DESCRIPTION
**Pull request by mightyguava**

### CODE REVIEW

File: misk-events/src/main/kotlin/misk/events/SpooledProducer.kt [hunk 1]
```diff
@@ -12,6 +12,20 @@ import misk.hibernate.Session
...
+   *
+   * @param session Session for the database transaction the event should be spooled in.
+   * @param groupRootId Entity group root ID for the event, used as the sharding key in the database,
+   *   and will be passed to the mapping function configured for the spool to look up a partition
+   *   key for the event.
+   */
   fun publish(session: Session, groupRootId: Gid<*, *>, topic: Topic, vararg event: Event)
+
+  /**
+   * Publishes an event to a sharded spool.
```
Comment (wesleyk) [inline misk-events/src/main/kotlin/misk/events/SpooledProducer.kt]: unsharded?

File: misk-events/src/main/kotlin/misk/events/SpooledProducer.kt [hunk 2]
```diff
@@ -12,6 +12,20 @@ import misk.hibernate.Session
...
+   *
+   * @param session Session for the database transaction the event should be spooled in.
+   */
   fun publishUnsharded(session: Session, topic: Topic, vararg event: Event)
```
Comment (wesleyk) [inline misk-events/src/main/kotlin/misk/events/SpooledProducer.kt]: worthwhile documenting the rest of the params?

File: misk-events/src/main/kotlin/misk/events/SpooledProducer.kt [hunk 3]
```diff
@@ -12,6 +12,20 @@ import misk.hibernate.Session
  * to the event stream, and are done so through a [Producer] transaction.
  */
 interface SpooledProducer {
+  /**
+   * Publishes an event to a sharded spool.
```
Comment (wesleyk) [inline misk-events/src/main/kotlin/misk/events/SpooledProducer.kt]: is it a single event or potentially many?

File: misk-events/src/main/kotlin/misk/events/SpooledProducer.kt (hunk 2 above)
Comment (mightyguava) [reply misk-events/src/main/kotlin/misk/events/SpooledProducer.kt]: No, I don't think so.

File: misk-events/src/main/kotlin/misk/events/SpooledProducer.kt (hunk 3 above)
Comment (mightyguava) [reply misk-events/src/main/kotlin/misk/events/SpooledProducer.kt]: Many, I'll update

**Comment by mightyguava** [comment]
I don't see what's failing on these travis builds

**Pull request merged**